src/ResourceManagement/ServerManagement/ServerManagement.Tests/OperationsTest.cs

```diff
@@ -0,0 +1,359 @@
+                    Assert.Equal(GatewayName, gateway.Name);
+                    Assert.Equal("microsoft.servermanagement/gateways", gateway.Type);
+                    Assert.Equal("centralus", gateway.Location);
+                    Assert.Equal(AutoUpgrade.On, gateway.AutoUpgrade);
+                    WriteLine(gateway.ToJson());
+
+                    // verify that we can get the gateway again.
+                    gateway = await client.Gateway.GetAsync(ResourceGroup, GatewayName);
+                    Assert.NotNull(gateway);
+                    Assert.Equal(GatewayName, gateway.Name);
+                    Assert.Equal("microsoft.servermanagement/gateways", gateway.Type);
+                    Assert.Equal("centralus", gateway.Location);
+                    Assert.Equal(AutoUpgrade.On, gateway.AutoUpgrade);
+                    WriteLine(gateway.ToJson());
+
+                    // update the gateway a bit.
+                    gateway = await client.Gateway.UpdateAsync(ResourceGroup, GatewayName, "centralus", autoUpgrade: AutoUpgrade.Off);
+                    Assert.Equal(AutoUpgrade.Off, gateway.AutoUpgrade);
+
+                    // get the gateway status
+                    gateway = await client.Gateway.GetAsync(ResourceGroup, GatewayName, GatewayExpandOption.Status);
+                    Assert.NotNull(gateway);
+
+                    // check for some extended properties
+                    Assert.NotNull(gateway.LatestPublishedMsiVersion);
+                    Assert.NotNull(gateway.ActiveMessageCount);
+                    WriteLine(gateway.ToJson());
+
+                    // Let's see how many gatways are in the subscription
+                    var gateways = await client.Gateway.ListAsync();
+                    Assert.NotNull(gateways);
+
+                    var found = false;
+                    foreach (var g in gateways)
+                    {
+                        found = true;
+                        WriteLine(String.Format("Found Gateway in subscription: {0}", g.Name));
+                    }
+
+                    Assert.True(found, "No gateways in collection?");
+                }
+                finally
+                {
+                    // remove gateway that we've created 
+                    RemoveGateway(client, GatewayName).Wait();
+                }
+
+                // make sure that the gateway is gone.
+                await Assert.ThrowsAsync<ErrorException>(() => client.Gateway.GetAsync(ResourceGroup, GatewayName));
+            }
+        }
+
+        [Fact]
+        public async Task NodeTest()
+        {
+            string GatewayName = "sdktest_gateway_105";
+
+            // ensure known state before starting.
+            await OneTimeInitialization();
+
+            using (var context = MockContext.Start("ServerManagement.Tests"))
+            {
+                var client = GetServerManagementClient(context);
+                GatewayResource gateway = null;
+
```

Comment:
It's .net coding guidelines.  line breaks in the middle of code blocks are generally not preferred  unless there is some desired logical break in the code (for example, before an inline comment) (they arguably make things less readable by lowering the number of lines on the screen.  Also, vertical space after ending curly brace, always.